client/src/app/unauthorized.component.test.tsx

```diff
@@ -0,0 +1,24 @@
+import { describe, expect, it } from 'vitest';
+
+import { render, screen, userEvent, waitFor } from '~/test/utils';
+
+import { Unauthorized } from './unauthorized.component';
+
+describe('Unauthorized', () => {
+  it('should render unauthorized component', () => {
+    render(<Unauthorized />);
+
+    expect(screen.getByRole('heading', { name: /unauthorized/i })).toBeInTheDocument();
+    expect(screen.getByText(/you do not have access to the requested page./i)).toBeInTheDocument();
+    expect(screen.getByRole('button', { name: /go back/i })).toBeInTheDocument();
+  });
+
+  it('should move one step backward from unauthorized page when clicks on the "Go Back" button', async () => {
+    const goBack = vi.fn();
+    render(<Unauthorized goBack={goBack} />);
+
+    userEvent.click(screen.getByRole('button', { name: 'Go Back' }));
```

Comment:
I thought I had previously linked to how to use the userEvent, please use it how the docs use it, not in some random stackoverflow, I got caught out by doing that. See https://testing-library.com/docs/user-event/intro/, it has an example. So there is setting up the user, then using the user to click things, it should be pretty self-explanatory, but if not, just ask.

One of the things is to await the click, that should probably negate the expect also having to be await'd, making it more linear, async stuff is hard.